delayed-state-rendering.md

```diff
@@ -0,0 +1,273 @@
+
+Here are several examples of how state files could use this
+
+### Simple example calling a jinja block:
+
+```salt
+# init.sls
+my_state1:
+  mod.fun1:
+    - do_something
+    - delayed_render:
+      - block: dependant_block_A
+
+{% delayed_block dependant_block_A %}
+my_state2:
+  mod.fun2:
+    - {{ data }}
+{% end_delayed_block %}
+```
+
+In this example, `my_state` indicates that `dependant_block_A` will be rendered and its states executed after it finishes. In this way, `delayed_render` can be thought of as a new kind of requisite. It is similar to `require_in` in that it helps determine an order of execution, and indicates what comes later. It is different, in that `delayed_render` determines what will be executed **immediately after** this state. It also does not simply point to another state. Instead, `delayed_render` points to a block of Jinja defined by the custom Jinja tag `delayed_block`. This block could contain any template code.
+
+Upon initial render of this state (as in a highstate or otherwise), the `delayed_block` Jinja tag effectively cuts all of the text in the block, and stores it in memory on the Salt master, and associates it with the original job. When `my_state1` finishes, the minion calls back to the master, indicating that the `delayed_block` `dependant_block_A` should be rendered, and any states within it executed, on the same minion.
+
+### Simple example calling an sls file:
+
+```salt
+# init.sls
+my_state1:
+  mod.fun1:
+    - do_something
+    - delayed_render:
+      - sls: my_state_file
+```
+
+```salt
+# my_state_file.sls
+my_state2:
+  mod.fun2:
+    - {{ data }}
+```
+
+This is a very similar example, except that instead of a `delayed_render` indicating a particular `delayed_block`, `delayed_render` specifies an entire sls file to render and execute when `my_state1` finishes.
+
+### Multiple delayed_renders
+
+```salt
+# init.sls
+my_state1:
+  mod.fun1:
+    - do_something
+    - delayed_render:
+      - sls: state_file1
+      - block: dependant_block_A
+      - sls: state_file2
+      - block: dependant_block_B
+```
+
+This state must be a precursor to many things. Since `delayed_render` contains a list, we can mark multiple blocks and sls files. This state indicates that two sls files and two blocks should be rendered after it finishes executing, and they should be rendered and executed in this particular order; `state_file1`, then `dependant_block_A`, etc.
+
+## Edge Cases
+
+Since the requisite and delayed render is so generalized, the user can potentially do complicated things.
```

Comment:
```suggestion
Since the requisite and delayed render is so generalized, the user can potentially do multiple step automation without user intervention after initial kick off.
```